### PR TITLE
feat(EditPolicy): RHICOMPL-1356 Alert user to new rule sets

### PR DESCRIPTION
**Pull request by akofink**
Builds on https://github.com/RedHatInsights/compliance-frontend/pull/1104

When selecting systems in the edit policy modal, alert users to new OS
minor versions and their new associated rule sets for tailoring.

image

Signed-off-by: Andrew Kofink <akofink@redhat.com>

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ rules:
       MemberExpression: 0
       ImportDeclaration: 1
       ObjectExpression: 1
+      ignoredNodes:
+      - TemplateLiteral
```
Comment (akofink) [inline .eslintrc.yml:56]: See https://github.com/babel/babel-eslint/issues/530#issuecomment-606355559

I've seen issues with template literals in the past but have been able to just remove them in favor of double quotes. Apparently some later version of babel-eslint has this fix, but I can't seem to upgrade it to resolve this.

File: .eslintrc.yml [hunk 2]
```diff
@@ -52,6 +52,8 @@ rules:
       MemberExpression: 0
       ImportDeclaration: 1
       ObjectExpression: 1
+      ignoredNodes:
+      - TemplateLiteral # https://github.com/babel/babel-eslint/issues/530
```
Comment (akofink) [inline .eslintrc.yml:56]: See https://github.com/babel/babel-eslint/issues/530#issuecomment-606355559

I've seen issues with template literals in the past but have been able to just remove them in favor of double quotes. Apparently some later version of babel-eslint has this fix, but I can't seem to upgrade it to resolve this.

File: src/SmartComponents/EditPolicy/EditPolicySystemsTab.js [hunk 3]
```diff
@@ -45,26 +58,34 @@ const EditPolicySystemsTab = ({ osMajorVersion }) => {
...
+                defaultFilter={ osMajorVersion && `os_major_version = ${osMajorVersion}` }
+                enableExport={ false }
+                remediationsEnabled={ false }
+            />
+            {newOsMinorVersions() && <Alert
+                variant="info"
+                isInline
+                title="You selected a system that has a release version previously not included in this policy."
+                actionLinks={
+                    <AlertActionLink onClick={ () => push({ ...location, hash: '#rules' }) }>Open rule editing</AlertActionLink>
```
Comment (akofink) [inline src/SmartComponents/EditPolicy/EditPolicySystemsTab.js:78]: Exact text pending UX review.

**Comment by skateman** [comment]
Works as expected, but when navigating to the rules tab after checking a new system there is no loading animation. This is probably out of the context of this PR.

**Review by skateman — APPROVED** [review]
The Seal of Approval

**Review by vkrizan — APPROVED** [review]
Works good.

UX-wise it could be improved to be more visible. Also the performance of a system selection is terrible :smile:.

**Comment by akofink** [comment]
>when navigating to the rules tab after checking a new system there is no loading animation

Reported as https://issues.redhat.com/browse/RHICOMPL-1651

File: src/SmartComponents/EditPolicy/EditPolicy.js [hunk 4]
```diff
@@ -198,6 +201,7 @@ export const EditPolicy = ({ route }) => {
                 <Tab eventKey='systems' title={ <TabTitleText>Systems</TabTitleText> }>
                     <EditPolicySystemsTab
                         osMajorVersion={ policy.osMajorVersion }
+                        policyOsMinorVersions={ [...new Set(policyProfiles.map(profile => profile.osMinorVersion))] }
```
[the remark this answers is not in the document]
Comment (akofink) [reply src/SmartComponents/EditPolicy/EditPolicy.js:204]: See https://github.com/RedHatInsights/compliance-frontend/pull/1115

**Pull request merged**